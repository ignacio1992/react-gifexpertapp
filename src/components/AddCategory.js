import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ( { setCategories } ) => {
    
    const [InputValue, setInputValue] = useState('');

    const handleInputChange = (e) =>{
            setInputValue( e.target.value );
    }
    //permite agregar nuevos elementos al arreglo
    const handleSubmit = (e) => {
        e.preventDefault();
        //console.log('Submit hecho');
        
        if ( InputValue.trim().length >2 ) {
            setCategories(anime =>[InputValue,...anime] );
            setInputValue('');   
        }
    }
    
    return (
        
        <form onSubmit ={handleSubmit}>
            <input 
             type ='text'
             value={InputValue}
             onChange={handleInputChange}
            />
        </form>

    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
