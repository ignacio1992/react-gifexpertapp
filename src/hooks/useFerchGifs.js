
import { useEffect, useState } from 'react'
import {getGifs} from '../helpers/getGifs'

export const useFerchGifs = ( category ) => {
   
    const [state, setState] = useState({
        data:[],
        loading: true
    });

    useEffect( ()=>{

        getGifs( category )
            .then( imgs =>{
                setTimeout(()=>{
                    setState({
                        data:imgs,
                        loading: false
                    })

                },1500);

            })
       
     },[category]) 

    return state; // {dadta:[], loading : true};
}



