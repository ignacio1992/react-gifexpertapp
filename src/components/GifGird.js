import React,{ } from 'react'
import { useFerchGifs } from '../hooks/useFerchGifs'
import { GifGridItem } from './GifGridItem';

export const GifGird = ({category}) => {

    const {data:images,loading} = useFerchGifs(category);

    return (
        <>
        <h3 className='animate__animated animate__fadeIn' > { category}</h3>

            { loading && <p className= 'animate__animated animate__flash'>loading...</p>}

        {<div className ='Card-grid'>
            
                {
                    images.map((img ) =>(
                        <GifGridItem
                         key= {img.id}
                         { ...img }
                         />
                    ))
                }
            </div>}
        </>
    )
}
