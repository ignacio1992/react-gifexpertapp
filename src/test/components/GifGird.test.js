import React from 'react';
import '@testing-library/jest-dom'
import {GifGird} from '../../components/GifGird'
import { shallow } from 'enzyme';
import { useFerchGifs } from '../../hooks/useFerchGifs';
jest.mock('../../hooks/useFerchGifs');

describe('pruebas en <GifGird/>', () => {

    const category = 'One Punch';
    //const wrapper = shallow (<GifGird category = { category }/>)

    test('debe mostrar correctamente ', () => {

        useFerchGifs.mockReturnValue({
            data: [],
            loading: true,
        });

        const wrapper = shallow(<GifGird category={category}/>);
        expect(wrapper).toMatchSnapshot(); 

    });

    test('debe de mostrar items cuando se cargan las imagenes <useFerchGifs/>  ', () => {

        const gifs = [{

            id: '123',
            url: 'https://HolaMundo.cl',
            title: 'Hola Mundo'

        }]

        useFerchGifs.mockReturnValue({
            
            data: gifs,
            loading: false

        });

        const wrapper = shallow(<GifGird category={category}/>);
        
        //expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe( gifs.length );
    })
    
  
})



