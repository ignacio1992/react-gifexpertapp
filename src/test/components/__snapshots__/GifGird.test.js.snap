// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifGird/> debe mostrar correctamente  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    loading...
  </p>
  <div
    className="Card-grid"
  />
</Fragment>
`;
